### PR TITLE
Implement nats.io message bus

### LINKED ISSUE #3: Implement nats.io message bus
Description
------------------------
As Carbonable I should be able to implement nats.io message bus
inorder to stream data to our users

In a file located at `internal/dispatcher/nats.go` implement interface *EventDispatcher* located in [dispatcher.go](../blob/main/internal/dispatcher/dispatcher.go) 

Additional Information
---------------------------
[Medium Difficulty  - Go - Medium Priority]

Acceptance Criteria
-------------------------
- [ ]  nats.io message bus

### PR DESCRIPTION
**Pull request by joaquinbarrientoscr**
Resolves: #3 
Implementation of EventDispatcher on NatsDispatcher.
Still work in progress. Submission for first revision.

### CODE REVIEW

File: internal/dispatcher/nats.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	"github.com/nats-io/nats.go"
+)
+
+type NatsDispatcher struct {
+	connection *nats.Conn
+	connError  error
+}
+
+func (nd *NatsDispatcher) Publish(topic string, data []byte) error {
+	nc := nd.connection
```
Comment (MartianGreed) [inline internal/dispatcher/nats.go]: For simplicity of reading we should write something like that : 

```go
err := nd.connection.Publish(topic, data)
if err != nil {
log.Error("error publishing to topic", "error", err, "topic", topic)
return error
}
log.Debug("Published to topic", "topic", topic, "payload", data)
return nil
```

what do you think ?

File: internal/dispatcher/nats.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	nc.Flush()
+	if err := nc.LastError(); err != nil {
+		log.Fatal(err)
+	} else {
+		log.Printf("Published [%s] : '%s'\n", topic, data)
+	}
+	return nil
+}
+
+func (nd *NatsDispatcher) Subscribe(topic string, handler nats.MsgHandler) error {
```
Comment (MartianGreed) [inline internal/dispatcher/nats.go:24]: be careful here on the interface, we may run into conflicts, i'll check it

File: internal/dispatcher/nats.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	nc := nd.connection
```
Comment (MartianGreed) [inline internal/dispatcher/nats.go]: same comment as above I think we should drop the nc var and inline the call to subscribe.

File: internal/dispatcher/nats.go (hunk 2 above)
Comment (joaquinbarrientoscr) [reply internal/dispatcher/nats.go:24]: yes, it seems nats.io uses it own struct for `MsgHandler`

File: internal/dispatcher/nats.go (hunk 1 above)
Comment (joaquinbarrientoscr) [reply internal/dispatcher/nats.go]: I think it makes sense, makes the code neater.

File: internal/dispatcher/nats.go (hunk 3 above)
Comment (joaquinbarrientoscr) [reply internal/dispatcher/nats.go]: absolutely!

**Review by MartianGreed — DISMISSED** [review]
lgtm !

**Review by MartianGreed — APPROVED** [review]

**Pull request merged**